pages/resources/quizzes/firstProgrammingLanguage/index.tsx

```diff
@@ -35,11 +37,58 @@ const initializeQuizViewState = {
 
 const FirstProgrammingLanguageQuiz = () => {
   const [stage, setStage] = useState(Stage.START);
+  const [quizResponseId, setQuizResponseId] = useState<number>();
   const [quizViewState, setQuizViewState] = useState<QuizViewState>(
     initializeQuizViewState
   );
+  const [userInput, setUserInput] = useState({
+    name: "",
+    email: "",
+  });
+
   const [triggerAnimation, setTriggerAnimation] = useState(true);
 
+  const [saveQuizResponse] = useMutation(UPSERT_CODING_LANGUAGE_QUIZ_RESPONSE, {
+    onCompleted: (data) => {
+      if (!quizResponseId) {
+        setQuizResponseId(
```

Comment:
why is it important to save this? how do you reuse this id later?

Reply:
@vithushan19 - can chime in here - the idea is that we write to the table when the user saves their information so we don't lose name / email, save the row id of what's inserted in state, and modify the same row through the same query when we get all the quiz responses